Modules/_decimal/_decimal.c

```diff
@@ -3241,11 +3267,31 @@ dec_format(PyObject *dec, PyObject *args)
             /* NUL fill character: must be replaced with a valid UTF-8 char
                before calling mpd_parse_fmt_str(). */
             replace_fillchar = 1;
-            fmt = dec_strdup(fmt, size);
-            if (fmt == NULL) {
+            fmt = fmt_copy = dec_strdup(fmt, size);
+            if (fmt_copy == NULL) {
                 return NULL;
             }
-            fmt[0] = '_';
+            fmt_copy[0] = '_';
+        }
+        /* Strip 'z' option, which isn't understood by mpd_parse_fmt_str().
+         * First, skip [[fill]align], since 'fill' itself may be 'z'.
+         * NOTE: fmt is always null terminated by PyUnicode_AsUTF8AndSize() */
+        char const *fmt_offset = size >= 2 && strchr("<>=^", fmt[1]) != NULL ?
+                                 fmt + 2 : fmt;
+        char *z_start = strchr(fmt_offset, 'z');
```

Comment:
It looks as though this is still too lenient: `format(Decimal("-0.000123"), ".2fz")` (for example) produces `0.00` instead of raising `ValueError`.

Reply:
Thank you-- I also realized that multi-byte fill character wasn't handled properly. These should be resolved.